services/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```
@@ -747,4 +752,36 @@ private String getPath(Node node) throws Exception {
 
     return nodePath;
   }
+
+  /**
+   * Search news with the given text
+   * @param text The text to be searched
+   */
+  public List<News> searchNews(String text, String lang) throws Exception {
+    SessionProvider sessionProvider = sessionProviderService.getSessionProvider(null);
+
+    Session session = sessionProvider.getSession(repositoryService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName(),
+            repositoryService.getCurrentRepository());
+    SearchContext context = new SearchContext(null, null);
+    context.lang(lang);
+    try {
+      List<News> newsList = new ArrayList<>();
+      List<ResultNode> resultNode = new ArrayList<>();
+      resultNode = newsSearchConnector.search(context, text, 0, 0, "relevancy", "desc");
+      resultNode.forEach(res -> {
+        try {
+          newsList.add(convertNodeToNews(res.getNode()));
+        } catch (Exception e) {
+          return;
```

Exceptions must not be swallowed. Please log an error with the exception:
```
LOG.error("Error while processing search result in News", e);
```